packages/cli/src/commands/screenshot-diff/screenshot-diff.command.ts

```diff
@@ -23,79 +26,198 @@ export class DiffError extends Error {
+      const headScreenshot = await readPng(headScreenshotFile);
+
+      const comparisonResult = compareImages({
+        base: baseScreenshot,
+        head: headScreenshot,
+        pixelThreshold: pixelThreshold ?? null,
+      });
+
+      totalMismatchPixels += comparisonResult.mismatchPixels;
+      totalComparedPixels += baseScreenshot.width * baseScreenshot.height;
+
+      logger.debug({
+        screenshotFileName,
+        mismatchPixels: comparisonResult.mismatchPixels,
+        mismatchFraction: comparisonResult.mismatchFraction,
+      });
+
+      await writeScreenshotDiff({
+        baseReplayId,
+        headReplayId,
+        screenshotFileName,
+        diff: comparisonResult.diff,
+      });
+
+      comparisonResults.push({
+        baseScreenshotFile,
+        headScreenshotFile,
+        comparisonResult,
+        outcome:
+          comparisonResult.mismatchFraction > threshold ? "fail" : "pass",
+      });
+    }
 
     await postScreenshotDiffStats(client, {
       baseReplayId,
       headReplayId,
       stats: {
-        width: baseScreenshot.width,
-        height: baseScreenshot.height,
-        mismatchPixels,
+        width: 0,
+        height: 0,
+        mismatchPixels: totalMismatchPixels,
       },
     });
 
-    if (mismatchFraction > threshold) {
-      logger.info("Screenshots do not match!");
+    const diffUrl = await getDiffUrl(client, baseReplayId, headReplayId);
+    logger.info(`View screenshot diff at ${diffUrl}`);
+
+    comparisonResults.forEach((result) => {
+      logComparisonResultMessage(
+        logger,
+        `${Math.round(
+          result.comparisonResult.mismatchFraction * 100
+        )}% pixel mismatch for screenshot ${basename(
+          result.headScreenshotFile
+        )} (threshold is ${Math.round(threshold * 100)}%)`,
+        result.outcome
+      );
+    });
+
+    // Check if individual screenshot mismatch is higher than the threshold.
+    const mismatchingScreenshots = comparisonResults.filter(
+      (result) => result.outcome == "fail"
+    );
+
+    if (mismatchingScreenshots.length) {
+      logger.info(
+        `Screenshots ${mismatchingScreenshots
+          .map((result) => basename(result.headScreenshotFile))
+          .sort()} do not match!`
+      );
       if (exitOnMismatch) {
         process.exit(1);
       }
-      throw new DiffError("Screenshots do not match!", {
-        baseReplayId,
-        headReplayId,
-        threshold,
-        value: mismatchFraction,
-      });
+      throw new DiffError(
+        `Screenshots ${mismatchingScreenshots.map((result) =>
+          basename(result.headScreenshotFile)
+        )} do not match!`,
+        {
+          baseReplayId,
+          headReplayId,
+          threshold,
+        }
+      );
+    }
+
+    const totalMismatchedFraction = totalMismatchPixels / totalComparedPixels;
+    const overallMismatchOutcome: ComparisonOutcome =
+      totalMismatchedFraction > threshold ? "fail" : "pass";
```

Comment:
I think doing the overall check is not needed (it's impossible to fail).

Reply:
ah yes, good point. AVG > threshold will be caught by the above individual screenshot checks https://github.com/alwaysmeticulous/meticulous-sdk/pull/132/commits/4ac49190537a6b06906cb85e0a2cf57b08e0b9c2